Use latest jom for test

xavier2k6 opened the pull request:
* Use latest jom (1.1.4) for test.
* **md5sum** = `4cd0e6fb721e063b24232bc9ecb6d130`
* https://download.qt.io/official_releases/jom/md5sums.txt
* https://mirrors.ukfast.co.uk/sites/qt.io/official_releases/jom/changelog.txt
* https://download.qt.io/official_releases/jom/?C=N;O=D

miurahr commented:
A change of jom version may bring the test failure which the case use jom. Please check it.

pzhlkj6612 commented:
Interesting, the entire Qt seems to be gone:

```
jom 1.1.4 - empower your cores

	D:\a\1\b\Qt\6.5.3\msvc2019_arm64\bin\jom.exe -f Makefile.Release
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -permissive- -Zc:__cplusplus -Zc:externConstexpr -O2 -MD -std:c++17 -utf-8 -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DWIN64 -DWIN64 -DNDEBUG -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -Itests\helloworld -I. -IQt\6.5.3\msvc2019_arm64\include -IQt\6.5.3\msvc2019_arm64\include\QtWidgets -IQt\6.5.3\msvc2019_arm64\include\QtGui -IQt\6.5.3\msvc2019_arm64\include\QtCore -Irelease -I/include -IQt\6.5.3\msvc2019_arm64\mkspecs\win32-arm64-msvc -Forelease\ @C:\Users\VSSADM~1\AppData\Local\Temp\main.obj.940.0.jom
main.cpp
	link /NOLOGO /DYNAMICBASE /NXCOMPAT /OPT:REF /OPT:ICF /INCREMENTAL:NO /SUBSYSTEM:WINDOWS "/MANIFESTDEPENDENCY:type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' publicKeyToken='6595b64144ccf1df' language='*' processorArchitecture='*'" /MANIFEST:embed /OUT:release\Hello.exe @C:\Users\VSSADM~1\AppData\Local\Temp\Hello.exe.940.10297.jom
main.obj : error LNK2001: unresolved external symbol __imp_qt_version_tag_6_5
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl QString::~QString(void)" (__imp_??1QString@@QEAA@XZ) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl QString::QString(char const *)" (__imp_??0QString@@QEAA@PEBD@Z) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl QApplication::QApplication(int &,char * *,int)" (__imp_??0QApplication@@QEAA@AEAHPEAPEADH@Z) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl QApplication::~QApplication(void)" (__imp_??1QApplication@@UEAA@XZ) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static int __cdecl QApplication::exec(void)" (__imp_?exec@QApplication@@SAHXZ) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __cdecl QWidget::show(void)" (__imp_?show@QWidget@@QEAAXXZ) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl QPushButton::QPushButton(class QString const &,class QWidget *)" (__imp_??0QPushButton@@QEAA@AEBVQString@@PEAVQWidget@@@Z) referenced in function main
main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl QPushButton::~QPushButton(void)" (__imp_??1QPushButton@@UEAA@XZ) referenced in function main
main.obj : error LNK2001: unresolved external symbol "public: __cdecl QString::~QString(void)" (??1QString@@QEAA@XZ)
main.obj : error LNK2001: unresolved external symbol "public: virtual __cdecl QApplication::~QApplication(void)" (??1QApplication@@UEAA@XZ)
main.obj : error LNK2001: unresolved external symbol "public: virtual __cdecl QPushButton::~QPushButton(void)" (??1QPushButton@@UEAA@XZ)
MSVCRT.lib(exe_winmain.obj) : error LNK2019: unresolved external symbol WinMain referenced in function "int __cdecl __scrt_common_main_seh(void)" (?__scrt_common_main_seh@@YAHXZ)
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Widgets.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Gui.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Core.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6EntryPoint.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
release\Hello.exe : fatal error LNK1120: 13 unresolved externals
jom: D:\a\1\b\Makefile.Release [release\Hello.exe] Error 1120

```

The Qt version update in "tests/test_install.py" looks unrelated to this PR, btw.

pzhlkj6612 commented:
I think you can revert the Qt version for now and see what will happen.

xavier2k6 commented:
> I think you can revert the Qt version for now and see what will happen.

Reverted to Qt 6.5.2 in "tests/test_install.py"

Changed azure matrix Qt version from 6.5.3 to 6.8.0 for testing purposes.

xavier2k6 commented:
azure pipelines seem to be unavailable/not running now?

pzhlkj6612 commented:
Hmm, force-pushing drops old commits and old logs... @xavier2k6, the commits in a PR can be squashed during merge [^1], so you can feel free to push anything.

```
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Widgets.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Gui.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6Core.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'
D:\a\1\b\Qt\6.5.3\msvc2019_arm64\lib\Qt6EntryPoint.lib : warning LNK4272: library machine type 'ARM64' conflicts with target machine type 'x64'

```

I think we need to look closer at `msvc2019_arm64` . Let's solve it in another PR.

[^1]: About merge methods on GitHub - GitHub Docs https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github

pzhlkj6612 commented:
@miurahr, I think we can embed the checksum of the ZIP package in our workflow and then check its integrity before extraction. The official list: https://download.qt.io/official_releases/jom/md5sums.txt

pzhlkj6612 commented:
See #895 for the checksum checking mechanism.

miurahr commented:
#895 updates jom version and superseded here.

pzhlkj6612 commented:
@xavier2k6, now you can create a merge commit with solving conflicts. @miurahr will squash commits before merge.

xavier2k6 commented:
@pzhlkj6612 I can re-base & modify last commit on my branch, wouldn't that be cleaner/easier? (I use github desktop)

pzhlkj6612 commented:
> I can re-base & modify last commit on my branch, wouldn't that be cleaner/easier? (I use github desktop)

Well, rebaseing and pushing to a existing PR always cause force-push, and then the previous CI run and code reviews, but since we now have no review, so you can force push. That's just my suggestions.

jdpurcell commented:
I think this will build now with the latest changes in `master`.

miurahr commented:
> I think this will build now with the latest changes in `master`.

@xavier2k6 please merge master branch to topic branch and push?
